<!DOCTYPE html>
<html>
  <head>
    <title>EMC Galaxy Training: Variant Analysis (DNA)</title>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    <style type="text/css">
      @import url(https://fonts.googleapis.com/css?family=Yanone+Kaffeesatz);
      @import url(https://fonts.googleapis.com/css?family=Droid+Serif:400,700,400italic);
      @import url(https://fonts.googleapis.com/css?family=Ubuntu+Mono:400,700,400italic);
      body { font-family: 'Droid Serif'; }
      h1, h2, h3 {
        font-family: 'Yanone Kaffeesatz';
        font-weight: normal;
      }
      .remark-code, .remark-inline-code { font-family: 'Ubuntu Mono'; }

      .red { color: #fa0000; }
      .blue { color: #0000aa; }
      .imagecenter { display: block; margin-left: auto; margin-right: auto; }
      .floatleft{float: left}
      .floatright{float: right}
      .small{font-size: 75%}
    </style>
  </head>
  <body>
    <textarea id="source">

class: center, middle

# Introduction to Data Processing and Variant Detection for NGS DNA Sequencing

.blue[
## EMC Galaxy Course

Youri Hoogstrate, David van Zessen, Saskia Hiltemann,
Andrew Stubbs
]
---
class: center, middle
# How does next-gen sequencing work?

.center[![:scale 75%](images/ngs.png)]

---
class: center, middle
## Instruments generate short reads that need to be mapped to the reference

.center[![:scale 85%](images/reads.png)]

---
class: center, middle
## High-level overview of NGS data processing

.center[![:scale 85%](images/sequencing.png)]

---

class: center, middle
## Aligned reads
In Galaxy, you can view your data in the built-in genome browser **Trackster**

.center[![:scale 85%](images/mapped_good.png)]

---

class: center, middle
## Challenge: distinguising variants from noise

.center[![:scale 75%](images/mapped_bad.png)]

.left[
**Possible reasons for a mismatch:**
- True SNV
- Error generated in library prep
- Base calling error
- Misalignment (mapping error)
- Error in reference genome
]
---
class: center, middle
## Genotyping
.left[
- What are the set of alleles at this locus? What are their frequencies?
- Genotypers begin with a model of prior knowledge about the likelihood (and types) of errors, and the likelihood of observing real variants
- Error models depend on sequencing technology
]

.center[
![:scale 50%](images/genotyping2.png)
![:scale 65%](images/genotyping.png)
]



---
class: center, middle
## What we know about NGS technology

.left[
- Relatively high per-base error rate
- Reads are higher quality in the middle than at the ends
- Some technologies are poor with homopolymers, GC rich
- Indels confuse alignment
- Sequence coverage is not uniform
- Alignments are probabilistic

![:scale 50%; float: right](images/indelrealignment.png)
**Quality Control**
- Local realignment
- Remove duplicate reads
- Filter low-quality reads
- Recalibrate base qualities
- Read trimming
]

---
.center[
## Quality Scores
]

**FastQ:** raw reads with per-base quality scores

.center[ ![:scale 80%](images/fastq.png) ]

.left[
.small[
Quality = Phred score + 33 (so that all characters are printable)  
Q = -10 log Pe (P = base-calling error probability)  
Q = 10: error rate 10%  
Q = 20: error rate 1%  
Q = 30: error rate 0.1%  
..  
]
![:scale 70% ;float: right](images/fastq_scoreflavours.png) 


**Different flavours**  
Sanger  
Solexa  
Illumina 1.3+  
Illumina 1.8+  
]



---
.center[
## Quality Control
]

**Tool:** FastQC

.center[ ![:scale 100%](images/fastqc_outputs.png) ]

---
class: center, middle
## Sequencing Depth

![:scale 80% ](images/depthofcoverage1.png) 

---
.center[
## Variant Callers
]

.left[
**Popular Tools**  
- SAMtools Mpileup
- GATK Unified Genotyper
- FreeBayes
- MAQ
- Varscan2 (practical)
]
All available in the Galaxy Tool Shed  
output: **VCF file**  


.center[ ![:scale 80% ](images/vcfformat.png) ]


---
.center[
## Variant Callers
]
Challenge: variants can have multiple representations

![:scale 90% ](images/variantcallers_comparison_giab.png) 

---
class: center
## Practical

Raw data (fastq files)

![:scale 5% ](images/arrowdown.png) 

QC with **FastQC**  
Map with **BWA**  
Visualize with **Trackster**  
Call variants with **Varscan2**  
Annotate variants with **ANNOVAR**  


.bottom[
Manual: https://www.bioinf-galaxian.erasmusmc.nl/galaxy
]
---
.center[
## Title
]
---

    </textarea>
    <script src="https://gnab.github.io/remark/downloads/remark-latest.min.js" type="text/javascript">
    </script>
    <script type="text/javascript">
        remark.macros.scale = function (percentage) {
            var url = this;
            return '<img src="' + url + '" style="width: ' + percentage + '" />';
            // Usage:
            //   ![:scale 50%](image.jpg)
            // Outputs:
            //   <img src="image.jpg" style="width: 50%" />
       };
       remark.macros.float = function (where) {
            var url = this;
            return '<img src="' + url + '" style="float: ' + where + '" />';
            // Usage:
            //   ![:float right](image.jpg)
            // Outputs:
            //   <img src="image.jpg" style="float: right" />
       };
      var slideshow = remark.create();
    </script>
  </body>
</html>
