\documentclass[11pt,a4paper]{article}
\usepackage[utf8]{inputenc}
\usepackage[english]{babel}

\usepackage{graphicx}
\usepackage[english]{babel}
\usepackage[margin=0.85in]{geometry}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{hyperref}
\bibliographystyle{plainnat}
\usepackage{setspace}

\input{../configure.tex}% Configurations should be changed in this file

\begin{document}
\title{ \textit{\institute}\text{ }Galaxy Training: RNA-Seq DGE analysis \\
{ \large This practical aims to familiarize you with the Galaxy RNA-Seq analysis, the FastQ format and data collections (pairs).}}

\author{ \authors }
\maketitle

%\newpage
%\tableofcontents
%\newpage

%\doublespacing

\section*{Introduction}
Due to the rapid development of Galaxy, screenshots and results may be out of date. If you experience something like this, please report it as a bug at \url{https://github.com/ErasmusMC-Bioinformatics/galaxy-courses/issues}.

\input{../Preparations.tex}% Preparations should describe how to access which server

% Module QC/QA:
\input{Galaxy_RNA-Seq_DGE_manual_Module_QC.tex}

% Module alignment:
\section{RNA-Seq: alignment}
To make more sense of the RNA-Seq data, we try to find the sequences back in the reference genome (mapping, aligning). The reference
genome should represent the most common sequence of the chromosomes of the human population. Please read the first paragraph (3 scentences) of the following url: \url{http://en.wikipedia.org/wiki/Reference_genome}
\begin{itemize}
	\item On how many individuals is hg19 based?
\end{itemize}
For RNA-Seq we need specialized RNA aligners, able to cope with gaps that originate from splicing. For this exercise we will make use of a tool called RNA-STAR. There are quite some of these aligners around. We will make use the tool ``\textit{\underline{HISAT2} A fast and sensitive alignment program}''. Load the aligner, select the following settings and leave the rest on default, and run an alignment for \textit{miR-23b (clean)} and \textit{control sample (clean)}. \textbf{Alignment is a computational very very heavy task so do NOT re-run or load it twice, or you have to give a treat:}\\
\includegraphics[width=\textwidth]{figures/alignment_01.png}\\
Please rename the \textit{HISAT2 on ...} to ``\textit{HISAT2 on miR-23b}'' and ``\textit{HISAT2 on control sample}''.
If the alignments do not have their \textbf{database} set, change it to hg19 as follows:\\
\includegraphics[width=\textwidth]{figures/alignment_02.png}\\
To get some general alignment statistics, run the tool ``\textit{\underline{Flagstat} tabulate descriptive stats for BAM datset}'' on \textit{HISAT2 on miR-23b}.
\begin{itemize}
	\item How many reads are multi-mapping?
\end{itemize}
For now we have only seen FastQ data and some summaries. To get an idea of what has been measured during the experiment, we can visualise the alignment. So, in the HISAT2 step we have been looking into hg19 where these sequences can be found in the reference genome, and this information is stored in those bam files. Start the built-in visualization Trackster at one of the alignments (make sure the dataset has \textbf{hg19} as database because we aligned to that):\\
\includegraphics[scale=0.55]{figures/alignment_03}\\
Give it a name, press \textit{Create} and you show a yellow bar, indicating that the BAM file is being prepared. This means that the bam file is being convert into a file format that the browser can visualize. \includegraphics[width=\textwidth]{figures/alignment_04.png}\\
When this job is done, and it looks like this, make sure you save it:\\
\includegraphics[width=\textwidth]{figures/alignment_05.png}\\
To add the other alignment, press the \textbf{[+]}-button:\\
\includegraphics[width=\textwidth]{figures/alignment_06.png}\\
Now select the other alignment and press \textit{Add} and save it again:\\
\includegraphics[width=\textwidth]{figures/alignment_07.png}\\
Remark that this is truncated data and you are supposed to see barely anything in here. Go to \textit{chr16}, to region \verb|chr16:15696870-15745667|.


% The singletons
%could be aligned separately in a single-end alignment, and could be added to the same alignment
%later on (but this is out of the scope of this tutorial). Go to Tophat2 and select Pair-end library. Make
%sure the reference genome is hg19. You have to run TWO alignments – one %for the quality trimmed
%“treat_” samples and one for the quality trimmed “ctrl_” samples.


%\bibliographystyle{natbib}
%\bibliographystyle{plainnat}

%\newpage

\vspace{-1.5em}
\bibliography{references}

\end{document}