\section{RNA-Seq: QC/QA}
The very first step of an RNA-Seq analysis is the quality control and quality assurance. The sequencing data is usually provided in FASTQ format, which reports the sequenced bases and also describes per base the quality reflecting the probability it was measured correctly. However, different encodings are being used. Galaxy contains tools that assist you in determining and dealing with these formats by converting them to \textit{fastqsanger}.

\subsection{Paired end data as `pairs' within galaxy}
\textbf{Note: using `pairs' for storing paired end reads in galaxy is experimental but will become the common way doing analysis on paired end data.} \\
This section will only show how to do this.

For this module, create a new history and load the Shared Data (\textit{\datalibrarydirrnaseqadvanced}) files ``ctrl\_small\_1.fq'' and ``ctrl\_small\_2.fq'' and ``treat\_small\_1.fq'' and ``treat\_small\_2.fq'' into your history. As you may expect from the file names already, \verb|_1| and \verb|_2| indicate that these files are paired end and belong together:\\
\includegraphics[scale=0.65]{figures/qc_01.png}\\
Hence, each pair of files belong to each other and we can treat them as pairs in galaxy. First you have to select the checkbox on top of the history ``\textit{Operations on multiple datasets}'', select the history items that belong to the pair of interest and then press the button \textit{For all selected...} to apply actions on multiple history items at once, select \textit{Build Dataset Pair}. Galaxy does not really know which one is the forward and which one is the reverse. Therefore make sure that \textbf{\_1 is forward} and \textbf{\_2 is reverse}. If galaxy does not choose the files correctly by it itself, use \text{Swap}. One of the samples has been treated with miR-23b and the other is a control. Make sure you use names that make this clear. After you have done this, you will see that you end up with 6 datasets in total, of which 2 pairs. To avoid confusion it is easier to hide the individual history items, leaving two items in the history:\\
\includegraphics[width=\textwidth]{figures/qc_02.png}\\
\subsection{Fastq, Fastqsanger and FastQC}
To obtain statistics on the data, find the tool ``\textit{\underline{FastQC} Read Quality reports}''. This tool, originally developed for DNA-Seq analysis, makes several summaries of the data and reports the summary in a HTML page. By default FastQC selects the individual fastq files for analysis. This is because FastQC does analysis just on one fastq file. When we choose to select one of our pairs, galaxy will run FastQC twice and merge the results into a new pair. You can choose whether you want to run them as pair or as single file seperately, but make sure you run all of them to get an impression of the data. In the following figure it is demonstrated how to use FastQC in combination with pairs:\\
\includegraphics[width=\textwidth]{figures/qc_03.png}\\
What follows is a quality report (raw data and HTML report) for each FASTQ file. Open the miR23b (forward) FastQC report that contains \textit{\ldots: Webpage \ldots} in its name which should look similar to this:\\
\includegraphics[scale=0.65]{figures/qc_04.png}\\
The galaxy history items are in the \textbf{fastq} format. This format is a general format that covers four specific subformats that differ in their quality encoding. Before we proceed with the next step it is important to understand the FASTQ encoding formats. They are described at the following website in section \textit{Encoding}:\\
\url{http://en.wikipedia.org/wiki/FASTQ\_format#Encoding}\\
In the FastQC report, scroll down to \textbf{Basic Statistics} and figure out what the \text{Encoding} is. This information is important because other tools want the specific subdatatype instead of `\textit{fastq}', so write the format as reported by FastQC down:\\
\verb|.......................................................|\\
The more or less standard encoding is \textit{fastqsanger}, which is the same encoding as \textit{Illumina 1.8} and \textit{Illumina 1.9}, also referred to as \textit{Illumina 1.8+}. To convert a fastq file into \textit{fastqsanger}, we use the tool ``\textit{\underline{FASTQ Groomer} converts between various FASTQ quality formats}''. In here, make sure the quality encoding you wrote down matches the one in the input field. Further, make sure you run it for each sample:\\
\includegraphics[width=\textwidth]{figures/qc_05.png}\\
This will add the groomed fastqsanger files to your history, with names that are not convenient. Please rename such that you can keep track of the sample name (miR23b/control) and, if you did not use pairs, whether the reads are forward or reverse.

If we go back to the FastQC webpage report, and look in section \textit{Per base sequence quality}, we see the average quality per base for all reads. The colors green, orange and red indicate whether the quality is considered good, okay, or bad. As you can see, the quality drops as the sequences get longer. It is important to realize that low quality bases will complicate alignment as well as SNP detection, because there will be more mismatches. To improve overall the base quality of the data, we would like to:
\begin{itemize}
	\item Trim the low quality bases from the ends
	\item Remove reads of which the average quality is too low
	\item Remove reads that are too short
\end{itemize}
A tool that covers all of this is ``\textit{\underline{Sickle} windowed adaptive trimming of FASTQ data}''. Because we have paired end data, we have to run it twice, once for miR-23b and once for the control. Run it with the following settings:\\
\includegraphics[width=\textwidth]{figures/qc_06.png}\\
After running a sickle analysis, rename the files 
\textit{Singletons from paired-end ...} and \textit{Paired-end output of Sickle on ...} to e.g.:
\begin{itemize}
	\item[] for miR-23b:
	\begin{itemize}
		\item miR-23b, singletons(clean)
		\item miR-23b\_1 (clean)
		\item miR-23b\_2 (clean)
	\end{itemize}
	\item[] for control sample:
	\begin{itemize}
		\item control sample, singletons(clean)
		\item control sample\_1 (clean)
		\item control sample\_2 (clean)
	\end{itemize}
\end{itemize}
If desired, you can hide the other results, such that you will get a history similar to:\\
\includegraphics[scale=0.55]{figures/qc_07.png}\\
As you can see, Sickle produces for every set of paired sequencing reads, a set of pairs and an extra file with ``singletons''.
\begin{itemize}
	\item What would singletons be?
\end{itemize}
To confirm that the base quality has improved, run the FastQC again on \textit{ miR-23b (clean) }, and take a look at it:
\begin{itemize}
	\item Has the \textit{Per base sequence quality} improved?
	\item Have the \textit{Per sequence quality scores} improved?
	\item Why has the \textit{sequence length distribution} changed?
\end{itemize}
FastQC also has a section \textit{Overrepresented sequences}, indicated in red with a huge list. Apart from that we are using a truncated artificial dataset, it often happens in RNA-Seq data that these sequences appear. As said before, this tool was orignally written for DNA-Seq data.
\begin{itemize}
	\item Could you think of a reason why sequences could be overrepresented in RNA-Seq data?
\end{itemize}
